commands/command_status.go

```diff
@@ -244,9 +244,25 @@ func jsonStagedPointers(ref string) {
 		ExitWithError(err)
 	}
 
+	scanner, err := lfs.NewPointerScanner()
```

Comment:
It looks like this is being created in order to pass it to the `blobInfoFrom` call on L257 below. This is fine, though I'd rather pass the `*lfs.PointerScanner` we have in the `statusCommand` top-level function, like this:

```diff
diff --git a/commands/command_status.go b/commands/command_status.go
index e74e26b5..bfda7f8e 100644
--- a/commands/command_status.go
+++ b/commands/command_status.go
@@ -30,11 +30,18 @@ func statusCommand(cmd *cobra.Command, args []string) {
 		scanIndexAt = git.RefBeforeFirstCommit
 	}
 
+	scanner, err := lfs.NewPointerScanner()
+	if err != nil {
+		scanner.Close()
+
+		ExitWithError(err)
+	}
+
 	if porcelain {
 		porcelainStagedPointers(scanIndexAt)
 		return
 	} else if statusJson {
-		jsonStagedPointers(scanIndexAt)
+		jsonStagedPointers(scanner, scanIndexAt)
 		return
 	}
 
@@ -44,14 +51,6 @@ func statusCommand(cmd *cobra.Command, args []string) {
 	if err != nil {
 		ExitWithError(err)
 	}
-
-	scanner, err := lfs.NewPointerScanner()
-	if err != nil {
-		scanner.Close()
-
-		ExitWithError(err)
-	}
-
 	Print("\nGit LFS objects to be committed:\n")
 	for _, entry := range staged {
 		switch entry.Status {
@@ -238,7 +237,7 @@ type JSONStatus struct {
 	Files map[string]JSONStatusEntry `json:"files"`
 }
 
-func jsonStagedPointers(ref string) {
+func jsonStagedPointers(s *lfs.PointerScanner, ref string) {
 	staged, unstaged, err := scanIndex(ref)
 	if err != nil {
 		ExitWithError(err)
@@ -247,6 +246,9 @@ func jsonStagedPointers(ref string) {
 	status := JSONStatus{Files: make(map[string]JSONStatusEntry)}
 
 	for _, entry := range append(unstaged, staged...) {
+		_, from, err := blobInfoFrom(s, entry)
+		// ...
+
 		switch entry.Status {
 		case lfs.StatusRename, lfs.StatusCopy:
 			status.Files[entry.DstName] = JSONStatusEntry{
```